Fix microseconds conversion in TestCrayAlpsReleaseTunables

shilpakodli opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Describe Bug or Feature
Two tests from TestCrayAlpsReleaseTunables were failing with error "ValueError: unconverted data remains: .621222".
In PTL, we are adding high resolution logging in all daemons logs. In test case, we are getting only till seconds of string in strptime as "%m/%d/%Y %H:%M:%S" and hence the error.

After fixing above issue, test started failing with error : AssertionError: 3.2769370079040527 not less than or equal to 3 : alps_release_jitter not working

#### Describe Your Change
Added microseconds formatting while converting string to date time.
Add math.floor to get the time difference to an integer value.

#### Link to Design Doc
<!--- If there is a design, link to it here: **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)** -->


#### Attach Test and Valgrind Logs/Output
[TestCrayAlpsReleaseTunables_after_fix.txt](https://github.com/openpbs/openpbs/files/4959829/TestCrayAlpsReleaseTunables_after_fix.txt)
[TestCrayAlpsReleaseTunables_before_fix.txt](https://github.com/openpbs/openpbs/files/4959830/TestCrayAlpsReleaseTunables_before_fix.txt)


<!--- Pull Request Guidelines: [Pull Request Guidelines](https://pbspro.atlassian.net/wiki/spaces/DG/pages/1187348483/Pull+Request+Guidelines) -->

bhroam commented on test/tests/functional/pbs_alps_release_tunables.py:
```diff
@@ -44,7 +44,7 @@
 def get_epoch(msg):
     # Since its a log message split on ';' to get timestamp
-    a = time.strptime(msg.split(';')[0], "%m/%d/%Y %H:%M:%S")
+    a = time.strptime(msg.split(';')[0], "%m/%d/%Y %H:%M:%S.%f")
```
Use LogUtil's convert_date_time().  This handles microseconds properly.  time's %f is broken.  You need to use datetime which convert_date_time() does.

shilpakodli commented:
@bhroam I have addressed your review comments. Please find latest execution logs.
[Latest_execution_logs.txt](https://github.com/openpbs/openpbs/files/5014099/Latest_execution_logs.txt)

I have changed math.ceil to math.floor because, test was failing with assertion error.

lisa-altair commented:
Hi @shilpakodli, one thing I do not understand is why things are being made to be integers.  
Previously there was no high resolution logging available, and so the float values of "alps_release_wait_time" and "alps_release_jitter" had to be compared against integer log values.  But now that there is high resolution logging, then it should be even easier to use the float values, right?

lisa-altair commented on test/tests/functional/pbs_alps_release_tunables.py:
```diff
@@ -151,10 +149,11 @@ def test_alps_release_jitter(self):
-                    time_current = get_epoch(data[1])
-                    self.assertLessEqual(time_current - time_prev, max_delay,
+                    time_current = self.get_epoch(data[1])
+                    self.assertLessEqual(math.floor(time_current - time_prev),
```
You shouldn't need math.floor here.  Just compare the high res times against the real max_delay time.  The logs should not be spread out more than max_delay.  However, it may be possible that if the system is a bit slow and the random number generated for jitter makes us equal to max_delay, then the logs can be logged a bit slower and thus be farther apart than max_delay.  In which case I think you can increase the alps_release_wait_time to workaround the slower system.

lisa-altair commented on test/tests/functional/pbs_alps_release_tunables.py:
```diff
@@ -96,12 +94,12 @@ def test_alps_release_wait_time(self):
                 fail_msg = "alps_release_wait_time not working"
                 self.assertGreaterEqual(time_current - time_prev,
-                                        math.ceil(arwt),
+                                        math.floor(arwt),
```
Now that we have high res timestamps, do we still need to make arwt into an integer?

shilpakodli commented:
@bhroam and @lisa-altair I have addressed your review comments. Please review latest code changes.
[Latest_execution_logs.txt](https://github.com/openpbs/openpbs/files/5069235/Latest_execution_logs.txt)

lisa-altair commented on test/tests/functional/pbs_alps_release_tunables.py:
```diff
@@ -3,49 +3,41 @@
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
 #
-# OpenPBS is distributed in the hope that it will be useful, but WITHOUT
```
This copyright/license header still looks wrong.

lisa-altair commented on test/tests/functional/pbs_alps_release_tunables.py:
```diff
@@ -151,10 +149,11 @@ def test_alps_release_jitter(self):
-                    time_current = get_epoch(data[1])
-                    self.assertLessEqual(time_current - time_prev, max_delay,
+                    time_current = self.get_epoch(data[1])
+                    self.assertLessEqual(int(time_current - time_prev),
```
I'm still not understanding why you are doing an int() here.  I must be missing something.  Can you please explain?  Thanks.

lisa-altair commented:
@shilpakodli Please attach latest test logs.

shilpakodli commented:
Attaching latest execution logs:
[Latest_execution_logs.txt](https://github.com/openpbs/openpbs/files/5082779/Latest_execution_logs.txt)

lisa-altair commented:
Thanks @shilpakodli looks good to me.
